This replicates the same disable_sse_win95.patch used for libsdl1.2

i.e. when running on Windows 95, never use SSE, as the OS doesn't support it

--- a/src/libFLAC/cpu.c
+++ b/src/libFLAC/cpu.c
@@ -100,6 +100,10 @@
 static const unsigned FLAC__CPUINFO_IA32_CPUID_EXTENDED_AMD_EXT3DNOW = 0x40000000;
 static const unsigned FLAC__CPUINFO_IA32_CPUID_EXTENDED_AMD_EXTMMX = 0x00400000;
 
+#if defined(_WIN32) && !defined(_WIN32_WCE)
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h> /* For Win95 check */
+#endif
 
 /*
  * Extra stuff needed for detection of OS support for SSE on IA-32
@@ -217,6 +221,15 @@
 #if defined FLAC__NO_SSE_OS
 			/* assume user knows better than us; turn it off */
 			info->data.ia32.fxsr = info->data.ia32.sse = info->data.ia32.sse2 = info->data.ia32.sse3 = info->data.ia32.ssse3 = false;
+#elif defined(_WIN32) && !defined(_WIN32_WCE)
+			OSVERSIONINFO versionInfo;
+			memset(&versionInfo, 0, sizeof(versionInfo));
+			versionInfo.dwOSVersionInfoSize = sizeof(versionInfo);
+			GetVersionEx(&versionInfo);
+
+			/* Is Win95? */
+			if(versionInfo.dwMajorVersion == 4 && versionInfo.dwMinorVersion == 0)
+				info->data.ia32.fxsr = info->data.ia32.sse = info->data.ia32.sse2 = info->data.ia32.sse3 = info->data.ia32.ssse3 = false;
 #elif defined FLAC__SSE_OS
 			/* assume user knows better than us; leave as detected above */
 #elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__DragonFly__) || defined(__APPLE__)
